### PR TITLE
Add support for NSCollectionView & NSTableView

### PR DESCRIPTION
**Pull request by OskarGroth**
Hello, thanks for an awesome framework!
This PR adds support for reloading `NSCollectionView` and `NSTableView` on macOS.
~~I've renamed and reorganized the existing `UIExtensions` to `Extensions` with categories for respective class.~~

gif

### CODE REVIEW

File: DifferenceKit.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -24,7 +24,10 @@
 		6B5B40AA211066EA00A931DB /* ArraySection.swift in Sources */ = {isa = PBXBuildFile; fileRef = 6B5B408A211066B300A931DB /* ArraySection.swift */; };
 		6B5B40AB211066EA00A931DB /* AnyDifferentiable.swift in Sources */ = {isa = PBXBuildFile; fileRef = 6B5B408B211066B300A931DB /* AnyDifferentiable.swift */; };
 		6B5B40AC211066EA00A931DB /* ElementPath.swift in Sources */ = {isa = PBXBuildFile; fileRef = 6B5B408D211066B300A931DB /* ElementPath.swift */; };
-		6B956B762110B25300DE3D29 /* UIKitExtension.swift in Sources */ = {isa = PBXBuildFile; fileRef = 6B5B4088211066B300A931DB /* UIKitExtension.swift */; };
+		75C6E50A214D66F80016493B /* UITableView+Reload.swift in Sources */ = {isa = PBXBuildFile; fileRef = 75C6E509214D66F80016493B /* UITableView+Reload.swift */; };
```
Comment (ra1028) [inline DifferenceKit.xcodeproj/project.pbxproj]: I prefer that `AppKitExtensions` and `UIKitExtensions` to the file names.

File: Sources/Extensions/NSCollectionView+Reload.swift [hunk 2]
```diff
@@ -0,0 +1,65 @@
+//
+//  NSCollectionView+Reload.swift
+//  DifferenceKit
+//
+//  Created by Oskar Groth on 2018-09-15.
+//  Copyright © 2018 Ryo Aoyama. All rights reserved.
+//
```
Comment (ra1028) [inline Sources/Extensions/NSCollectionView+Reload.swift]: [Nit] Please remove all the header comments.

File: Sources/Extensions/NSCollectionView+Reload.swift [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+#if os(macOS)
+import AppKit
+
+@available(OSXApplicationExtension 10.11, *)
+public extension NSCollectionView {
+
```
Comment (ra1028) [inline Sources/Extensions/NSCollectionView+Reload.swift]: [Nit] Please remove unneeded empty line.

File: Sources/Extensions/NSTableView+Reload.swift [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+                reloadData(forRowIndexes: IndexSet(changeset.elementInserted.map { $0.element }), columnIndexes: IndexSet(changeset.elementInserted.map { $0.section }))
+            }
+            
+            for (source, target) in changeset.elementMoved {
+                moveRow(at: source.element, to: target.element)
+            }
+            endUpdates()
+        }
+    }
+
```
Comment (ra1028) [inline Sources/Extensions/NSTableView+Reload.swift]: ditto

File: Sources/Extensions/UITableView+Reload.swift [hunk 5]
```diff
@@ -33,7 +41,7 @@ public extension UITableView {
             setData: setData
         )
     }
-
+
```
Comment (ra1028) [inline Sources/Extensions/UITableView+Reload.swift]: [Nit] Please revert all the white space changes.

**Comment by ra1028** [comment]
Could you add a simple example app?

**Comment by OskarGroth** [comment]
Cleaned up PR and added example project, see gif at top 😄

**Comment by dispatchqueue** [comment]
@OskarGroth @ra1028 I'm having an issue with the sample. When you click Shuffle button, the data you see in the table view is not same as the collection view although they are using the same array:

image

The collection view works well. If you add this code after shuffling the array, the table view will be reloaded and show correct data after 1s:

```swift
dataInput.shuffle()

DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
    self.tableView.reloadData()
    self.collectionView.reloadData()
}
```

**Pull request merged**